src/middlewared/middlewared/plugins/container/image.py

```diff
@@ -68,10 +69,15 @@ def pull(self, job, pool, image):
                 return snapshot_name
 
         self.middleware.call_sync('container.ensure_datasets', pool)
+        self.middleware.call_sync(
+            'pool.dataset.create_impl',
+            CreateImplArgs(
+                name=dataset_name, ztype='FILESYSTEM', create_ancestors=True,
+            )
+        )
         mountpoint = self.middleware.call_sync(
```

Comment:
There is no reason to query the dataset to get the mountpoint. It can be deduced without querying it.